.github/workflows/pull_request_check.yml

```diff
@@ -0,0 +1,71 @@
+name: Perform unit test on the new pull requests.
+
+on:
+  pull_request:
+    types: [opened]
+
+jobs:
+  project-setup-and-run-tests:
+    runs-on: ubuntu-latest
+
+     # Service containers to run with `runner-job`
+    services:
+      # Label used to access the service container
+      postgres:
+        # Docker Hub image
+        image: postgres
+        # Provide the password for postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          # Maps tcp port 5432 on service container to the host
+          - 5432: 5432
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+        
+      - name: Setup Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.9"
+
+      - name: Install python requirements
+        run: |
+          pip install -U pip
+          pip install poetry
+          poetry install --no-dev
+        env:
+          POETRY_VIRTUALENVS_CREATE: false
+
+      - name: Setup the database before running migrations
+          run: |
```

Comment:
There are some indention issues here, which makes the file invalid. You can see the error under the Action tab too: https://github.com/Oceans-1876/challenger-api/actions

One good way to test and debug actions locally is this package: https://github.com/nektos/act
It doesn't catch all the issues that might come up in an actual run, but can help with many others.

Reply:
Yeah, that was one thing I noticed, and was going to ask you in today's meet. Let me look into it and will try it out.

Reply:
I installed the `act`  package, but it seems to utilize a lot of your computer resources.